### PR TITLE
add addon resource test

### PR DESCRIPTION
**Pull request by songleo**
Add resource requirement e2e case for  endpoint-monitoring operator and metrics collector
https://github.com/open-cluster-management/backlog/issues/5764
Signed-off-by: Song Song Li <ssli@redhat.com>
/hold

### CODE REVIEW

**Comment by songleo** [comment]
Depend on https://github.com/open-cluster-management/endpoint-metrics-operator/pull/95

File: pkg/utils/mco_deploy.go [hunk 1]
```diff
@@ -629,6 +629,65 @@ func CheckMCOAddon(opt TestOptions) error {
...
+
+	endpointObsOperatorRes := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("600Mi"),
+		},
+		Requests: corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("100Mi"),
+		},
```
Comment (clyang82) [inline pkg/utils/mco_deploy.go:673]: Do we really need to check it? Can you imagine any cases that lead to it is not equal to default value?

Comment (songleo) [reply pkg/utils/mco_deploy.go:673]: check it just to ensure we have defined resource here: https://github.com/open-cluster-management/multicluster-observability-operator/blob/main/manifests/endpoint-observability/operator.yaml#L22

Comment (songleo) [reply pkg/utils/mco_deploy.go:673]: as you say. we never change this value. so I think we can remove this.

File: pkg/tests/observability_addon_test.go [hunk 2]
```diff
@@ -40,8 +40,17 @@ var _ = Describe("Observability:", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(enable).To(Equal(true))
 
-			By("Check ObservabilityAddon is created if there's managed OCP clusters on the hub")
+			By("Check addon resource requirement")
+			res, err := utils.GetMCOAddonSpecResources(testOptions)
```
Comment (clyang82) [inline pkg/tests/observability_addon_test.go:44]: Can you move this test out of `[P2][Sev2][Observability] Modifying MCO cr to disable observabilityaddon (addon/g0)` and add a new one with `[Integration]` metadata added?

Comment (songleo) [reply pkg/tests/observability_addon_test.go:44]: thanks, I will move it out.

**Pull request merged**